output/s3/s3_output.go

```diff
@@ -0,0 +1,190 @@
+package s3
+
+import (
+	"bytes"
+	"crypto/sha512"
+	"encoding/xml"
+	"fmt"
+	"hash"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"path"
+	"time"
+
+	"github.com/rlmcpherson/s3gof3r"
+	"github.com/spacemonkeygo/errors"
+	"github.com/spacemonkeygo/errors/try"
+	"polydawn.net/repeatr/def"
+	"polydawn.net/repeatr/lib/guid"
+	"polydawn.net/repeatr/output"
+	"polydawn.net/repeatr/output/tar2"
+)
+
+const Type = "s3"
+
+var _ output.Output = &Output{} // interface assertion
+
+/*
+	Amazon S3 silos are used by this transport in a very MVP-oriented way:
+	tarballs streamed in and out, because we know how to preserve file
+	attributes in a widely-understood way by doing this.
+
+	This IO system happens to share the same hash-space as the Tar IO system,
+	and may thus safely share a cache with Tar IO systems.
+
+	Working with S3 permissions: good luck; be advised that in many configurations,
+	accounts may be able to write data which they then instantly lose permission to
+	manage or even read back; this will result in permission denied errors from this
+	system, and may result in dirty state left behind (because we have to permission
+	to clean it up once we've created this, and cannot discover this until it's
+	too late!).  PRs for improving this behavior deeply welcome; this author
+	is not an S3/AWS/IAM professional.
+
+	Try the docs here: https://docs.aws.amazon.com/AmazonS3/latest/dev/example-bucket-policies.html --
+	these describe attaching policies to a bucket that can grant permissions to a user account.
```

Comment:
This is excellent, and I'd like to see either of these added:

1. Parts of the [PR submission comment](https://github.com/polydawn/repeatr/pull/42#issue-72183745) copied into a comment on the s3 input struct, or
1. Just add a link to said comment, until we get our act together documentation-wide

Thus hopefully alleviating "how does this work why no docs."